### PR TITLE
TF-4524 Fix blank plain-text email caused by backslash-hex false-positive in HTML sanitizer 

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#4524 

## Problem

Plain-text emails containing `\XX` patterns — such as PHP/Go namespace paths like `\Sabre\DAV\Exception\Forbidden` — were rendered as a completely blank white screen.

Root cause was a four-level chain:

1. `StandardizeHtmlSanitizingTransformers` (an HTML sanitizer) was incorrectly placed first in the `textPlain` pipeline — plain text is not HTML.
2. `NodeSanitizer._isEncodedJs()` in `dart-neats` uses `RegExp(r'\\[0-9a-f]{2}')` to detect unicode-escape obfuscation. This matches **any** single `\XX` pair, so `\DA` in `\Sabre\DAV` triggers it.
3. `_shouldStripText()` calls `node.remove()` → entire text node deleted.
4. `body.innerHtml` becomes empty → blank screen.

## Solution

Rebuilt the `textPlain` transformer pipeline with three dedicated layers:

| Layer | Transformer | Role |
|-------|-------------|------|
| 1 | `SanitizeAutolinkHtmlTransformers` | HTML-escape all plain text before any parsing; autolink URLs |
| 2 | `SanitizePlainTextHtmlOutputTransformer` *(new)* | Allow only `<a>` tags with safe `href` (validated via `Uri.tryParse`) and attribute allowlist; unwrap everything else |
| 3 | `PersistPreformattedTextTransformer` | Preserve line breaks and spacing |

The HTML sanitizer (`StandardizeHtmlSanitizingTransformers`) is removed from the plain-text path entirely.

It remains in the HTML path where it belongs.

## Related

- Need merge: https://github.com/linagora/dart-neats/pull/3

## Demo

Screenshot 2026-05-15 at 13 35 08

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed plain-text emails becoming blank when containing certain backslash-hex escape patterns.

* **New Features**
  * Enhanced HTML sanitization for plain-text emails with improved anchor tag validation, allowing only safe URL schemes (http, https, mailto) while removing dangerous attributes.

* **Tests**
  * Added comprehensive test coverage for HTML transformation and email content sanitization across various scenarios and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/linagora/tmail-flutter/pull/4527)

### CODE REVIEW

**Review by tddang-linagora — DISMISSED** [review]

File: contact/pubspec.lock [hunk 1]
```diff
@@ -1051,8 +1051,8 @@ packages:
     dependency: transitive
     description:
       path: sanitize_html
-      ref: support_mail
-      resolved-ref: "21f9da9bc177f5f29bc871d70f0b5f5fa2085d89"
+      ref: "fix/tighten-unicode-escape-regex"
+      resolved-ref: fefbb2dbdb63fa994ba28bd4551d4447dc7c60db
```
Comment (chibenwa) [inline contact/pubspec.lock]: Where is the PR for this fix ?

File: scribe/pubspec.lock [hunk 2]
```diff
@@ -1050,8 +1050,8 @@ packages:
     dependency: transitive
     description:
       path: sanitize_html
-      ref: support_mail
-      resolved-ref: "21f9da9bc177f5f29bc871d70f0b5f5fa2085d89"
+      ref: "fix/tighten-unicode-escape-regex"
+      resolved-ref: fefbb2dbdb63fa994ba28bd4551d4447dc7c60db
```
Comment (chibenwa) [inline scribe/pubspec.lock]: Why do we duplicate those pubspec ?

It seems like a maintenance hell

Please factorize: we shall have version declared only once in the entire project !

File: contact/pubspec.lock (hunk 1 above)
Comment (dab246) [reply contact/pubspec.lock]: I mentioned it in the PR description. Please check it out here: https://github.com/linagora/dart-neats/pull/3

File: core/test/utils/html_transform_text_plain_test.dart [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+      });
+
+      test('SHOULD prevent inline event handler from executing — tag becomes escaped text', () {
+        // <div onclick=...> in plain text is HTML-escaped by Layer 1, so no
+        // actual <div> element is produced and onclick cannot fire.
+        expect(
+          transform('<div onclick="steal()">Click me</div>'),
+          isNot(contains('<div ')),
+        );
+      });
```
[the remark this answers is not in the document]
Comment (dab246) [reply core/test/utils/html_transform_text_plain_test.dart:126]: Is this comment done?

File: test/features/email/data/local/html_analyzer_text_plain_test.dart [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+      test('SHOULD block <script> tag embedded in plain text email', () async {
+        expect((await transformPlain(TextPlainEmailCorpus.scriptTag)).content, isNot(contains('<script>')));
+      });
+
+      test('SHOULD prevent <img> pixel tracker from executing — tag becomes escaped text', () async {
+        // When <img> appears in a plain-text email, Layer 1 HTML-escapes its
+        // delimiters (< → &lt;, > → &gt;). No actual <img> element is produced
+        // so onerror cannot fire.
+        expect((await transformPlain(TextPlainEmailCorpus.imgTracker)).content, isNot(contains('<img ')));
+      });
```
[the remark this answers is not in the document]
Comment (dab246) [reply test/features/email/data/local/html_analyzer_text_plain_test.dart:111]: Is this comment done?

File: scribe/pubspec.lock (hunk 2 above)
Comment (dab246) [reply scribe/pubspec.lock]: You are right, this is a well-known maintenance pain in Dart/Flutter monorepos. Unlike Android's `libs.versions.toml`, Dart historically had no workspace-level version catalog, so each package had to redeclare git dependencies independently — a change to `sanitize_html`'s ref meant updating every sub-package lock file separately.

We have created a separate branch [migrate-dart-pub-workspaces](https://github.com/linagora/tmail-flutter/pull/4530) that implements Dart pub workspaces (available since Dart 3.5, which this project already uses):

- A single `pubspec.lock` at the workspace root replaces all 11 sub-package lock files
- `sanitize_html` and other git dependencies are declared once in root `dependency_overrides` and automatically propagate to all workspace members
- Updating a git ref now requires a one-line change in a single file

This PR is kept independent to avoid expanding the current bugfix scope. Happy to merge it shortly after this one.

Comment (chibenwa) [reply scribe/pubspec.lock]: Thanks

**Review by hoangdat — DISMISSED** [review]
